server/datastore/mysql/query_results_test.go

```diff
@@ -44,79 +43,125 @@ func testGetQueryResultRows(t *testing.T, ds *Datastore) {
 
 	mockTime := time.Now().UTC().Truncate(time.Second)
 
-	// Insert 2 Result Rows for Query1 and 1 empty data row
-	resultRows := []*fleet.ScheduledQueryResultRow{
+	// Insert Result Rows for Query1
+	query1Rows := []*fleet.ScheduledQueryResultRow{
 		{
 			QueryID:     query.ID,
 			HostID:      host.ID,
 			LastFetched: mockTime,
-			Data: json.RawMessage(
-				`{"model": "USB Keyboard", "vendor": "Apple Inc."}`,
-			),
-		},
-		{
-			QueryID:     query.ID,
-			HostID:      host.ID,
-			LastFetched: mockTime,
-			Data: json.RawMessage(
-				`{"model": "USB Mouse", "vendor": "Logitech"}`,
-			),
+			Data:        nil,
 		},
 		{
 			QueryID:     query.ID,
 			HostID:      host.ID,
 			LastFetched: mockTime,
-			Data:        nil,
+			Data: ptr.RawMessage([]byte(`{
+				"model": "USB Keyboard",
+				"vendor": "Apple Inc."
+			}`)),
 		},
 	}
-
-	err := ds.SaveQueryResultRows(context.Background(), resultRows)
+	err := ds.SaveQueryResultRows(context.Background(), query1Rows)
```

Comment:
Please remind me, why are we not using `ds.OverwriteQueryResultRows` here instead of the testing method `ds.SaveQueryResultRows`?

Reply:
looking again, it was somewhat of a convenience func.  I replaced any usage I could find with `Overwrite`